google/cloud/aiplatform/vizier/pyvizier/automated_stopping.py

```diff
@@ -0,0 +1,79 @@
+"""Convenience classes for configuring Vizier Early-Stopping Configs."""
+import copy
+from typing import Union
+
+import attr
+
+from google.cloud.aiplatform.compat.types import study as study_pb2
+
+AutomatedStoppingConfigProto = Union[
+    study_pb2.StudySpec.DecayCurveAutomatedStoppingSpec,
+    study_pb2.StudySpec.MedianAutomatedStoppingSpec,
+]
+
+
+@attr.s(frozen=True, init=True, slots=True, kw_only=True)
```

Comment:
Please run a global replace 
attr.s -> attr.define 
attr.ib -> attr.field

So we can take advantage of the newest features.